libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1919,6 +1934,14 @@ std::optional<std::pair<PeriodData, std::vector<std::shared_ptr<Vote>>>> PbftMan
     return std::nullopt;
   }
 
+  // Check reward votes
+  if (!vote_mgr_->checkRewardVotes(period_data.pbft_blk)) {
+    LOG(log_er_) << "Failed verifying reward votes. Disconnect malicious peer " << node_id.abridged();
+    sync_queue_.clear();
+    net->getSpecificHandler<network::tarcap::PbftSyncPacketHandler>()->handleMaliciousSyncPeer(node_id);
+    return std::nullopt;
+  }
+
```

Comment:
I would move this check before `// Check cert votes validation` as it should be cheaper

Reply:
Fixed